src/endpoint/ingress.cpp

```diff
@@ -190,85 +117,87 @@ void endpoint::Ingress::manage_entering_tcp_packets(void * mngmnt_args) {
                         INT_TO_IP_C_STR(headers.first.saddr),
                         headers.second.source,
                         INT_TO_IP_C_STR(headers.first.daddr),
-                        headers.second.dest, ttl, 0);
-        unsigned char formatted_pkt[args->pkt_len + SFC_HDR];
-        unsigned char* p = formatted_pkt;
-        utils::sfc_header::SFCUtilities::prepend_header((unsigned char*)args->pkt,
-                                                        args->pkt_len,
-                                                        flh, p);
-
-        printf("next ip %s\n", path[0].get_address().c_str());
-        printf("next port %d\n", next_port);
-        client::udp::ClientUDP().send_and_wait_response(formatted_pkt,
-                                                        args->pkt_len + SFC_HDR,
+                        headers.second.dest, static_cast<uint16_t>(ttl), 0);
+        std::vector<unsigned char> formatted_pkt(
+                static_cast<unsigned long>(args->pkt_len + SFC_HDR));
+        unsigned char *p = &formatted_pkt[0];
+        utils::sfc_header::SFCUtilities::prepend_header(
+                (unsigned char *) args->pkt,
+                static_cast<size_t>(args->pkt_len),
+                flh, p);
+
+        client::udp::ClientUDP().send_and_wait_response(p,
+                                                        static_cast<size_t>(
+                                                                args->pkt_len +
+                                                                SFC_HDR),
                                                         path[0].get_address().c_str(),
                                                         next_port);
     } else {
         LOG(ldebug, "no route available, discarding packages");
     }
-    delete(args->pkt);
+    delete (args->pkt);
     free(args);
 }
 
-void endpoint::Ingress::manage_pkt_from_chain(void * mngmnt_args) {
+void endpoint::Ingress::manage_pkt_from_chain(void *mngmnt_args) {
     LOG(ldebug, "manage packet from chain");
-    auto args = (server::udp::udp_pkt_mngmnt_args *)mngmnt_args;
+    auto args = (server::udp::udp_pkt_mngmnt_args *) mngmnt_args;
 
     sendto(args->socket_fd,
            ACK,
            ACK_SIZE,
            0,
-           reinterpret_cast<struct sockaddr*>(&args->client_address),
+           reinterpret_cast<struct sockaddr *>(&args->client_address),
            sizeof(args->client_address));
 
 
-
     struct sfc_header header =
-            utils::sfc_header::SFCUtilities::retrieve_header((uint8_t*)args->pkt);
+            utils::sfc_header::SFCUtilities::retrieve_header(
+                    (uint8_t *) args->pkt);
 
     struct iphdr ip_hdr =
             utils::PacketUtils::retrieve_ip_header(
-                    (unsigned char*)(args->pkt + SFC_HDR));
+                    (unsigned char *) (args->pkt + SFC_HDR));
     // TODO think something more extensible to support more protocols
```

Comment:
Did we complete this TODO?

Reply:
Never, I hope we will only support TCP and UDP. This TODO it is for posterity.

Reply:
Fair enough